src/runtime/chan.go

```diff
@@ -37,11 +37,78 @@ func chanDebug(ch *channel) {
 	}
 }
 
+// blockchain is a chain of blocked operations
+type blockchain struct {
+	// next blocked operation
+	next *blockchain
+
+	// t is the task to wake up
+	// the pointer state holds the data buffer unless this is a select
+	// the data field should be set to zero if this is an rx op when the channel closes
+	t *task
+
+	// s is the channel select state corresponding to this channel operation on a select statement
+	// if nil, just use the data in the task
```

Comment:
When exactly would this be nil? Just from this comment it sounds rather arbitrary.